kit/src/components/message/mod.rs

```diff
@@ -369,9 +370,57 @@ fn ChatText(cx: Scope<ChatMessageProps>) -> Element {
 
 pub fn markdown(text: &str) -> String {
     let txt = text.trim().replace(' ', "&nbsp;"); // need to do this else leading whitespaces are ignored
-    let parser = pulldown_cmark::Parser::new(&txt);
-    // Write to a new String buffer.
+
+    let mut options = Options::empty();
+    options.insert(Options::ENABLE_STRIKETHROUGH);
+
+    let text_lines: Vec<&str> = txt.split('\n').collect();
     let mut html_output = String::new();
-    pulldown_cmark::html::push_html(&mut html_output, parser);
-    html_output.replace("<p>", "").replace("</p>", "")
+    let mut in_paragraph = false;
+    let mut in_code_block = false;
+    for line in text_lines {
+        let parser = pulldown_cmark::Parser::new_ext(line, options);
+        for event in parser {
+            match event {
+                pulldown_cmark::Event::Start(Tag::Paragraph) => {
+                    in_paragraph = true;
+                    html_output.push_str("<p>");
+                }
+                pulldown_cmark::Event::End(Tag::Paragraph) => {
+                    in_paragraph = false;
+                }
+                pulldown_cmark::Event::Text(t) => {
+                    let txt: pulldown_cmark::CowStr<'_> = if in_paragraph {
+                        t.replace("\n\n", "<br/>").into()
+                    } else {
+                        t
+                    };
+                    pulldown_cmark::html::push_html(
+                        &mut html_output,
+                        std::iter::once(pulldown_cmark::Event::Text(txt)),
+                    );
+                }
+                pulldown_cmark::Event::Start(pulldown_cmark::Tag::CodeBlock(code_block_kind)) => {
+                    in_code_block = true;
+                    match code_block_kind {
+                        CodeBlockKind::Fenced(language) => html_output
+                            .push_str(&format!("<pre><code class=\"language-{}\">", language)),
+                        _ => html_output.push_str("<pre><code class=\"language-rust\">"),
```

Comment:
Might not be best to default to rust for this portion. Usually default would be something generic like text. What are all the options for `CodeBlockKind::Fenced`? probably could do something like 

```rust
let output = match code_block_kind {
    CodeBlockKind::Fenced(language) => language.to_string(), //This is assuming `Display` or `ToString` is implemented
    _ => "text".to_string(), //Assuming `language-text` or something along the like is available
};
html_output.push_str(&format!("<pre><code class=\"language-{}\">", language));
```

Note that the above snippet is just an example since im only guessing at what the output might be there right now